htdocs/product/list.php

```diff
@@ -1200,6 +1245,38 @@
 			if (!$i) $totalarray['nbfield']++;
 		}
 
+
+		// Multiprices
+		if ($conf->global->PRODUIT_MULTIPRICES){
+			foreach ($arraypricelevel as $key => $value)
+			{
+				if (!empty($arrayfields['p.sellprice'.$key]['checked']))
+				{
+					$resultp = "SELECT p.rowid, p.fk_product, p.price, p.price_ttc, p.price_level, p.date_price";
+					$resultp .= " FROM ".MAIN_DB_PREFIX."product_price as p";
+					$resultp .= " WHERE fk_product = ".$obj->rowid;
+					if (!empty($conf->global->PRODUIT_MULTIPRICES)) $resultp .= " AND p.price_level = ".$key;
+					$resultp .= " ORDER BY p.date_price DESC, p.rowid DESC, p.price_level ASC";
+					$resultp = $db->query($resultp);
+					if ($resultp)
+					{
+						$objp = $db->fetch_object($resultp);
+						print '<td class="right nowraponall">';
+						if ($obj->tosell)
```

Comment:
Why don't you set this test higher to avoid to make the select on product_price if obj->tosell is empty ?

Reply:
@eldy Yes for optimize i change the code